// Copyright 2024 Sonardyne

// Permission is hereby granted, free of charge, to any person obtaining a copy
// of this software and associated documentation files (the “Software”), to deal
// in the Software without restriction, including without limitation the rights
// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
// copies of the Software, and to permit persons to whom the Software is
// furnished to do so, subject to the following conditions:

// The above copyright notice and this permission notice shall be included in
// all copies or substantial portions of the Software.

// THE SOFTWARE IS PROVIDED “AS IS”, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
// SOFTWARE.

syntax = "proto3";

import "sonardyne-public/idl/common/result.proto";
import "sonardyne-public/idl/common/uid.proto";

package sonardyne.api.pub.configuration;

message AidingState{
  enum AidingStateEnum {
    UNKNOWN                  = 0;
    ENABLED                  = 1;
    DISABLED                 = 2;
  }

  AidingStateEnum value = 1; // The state of the aiding
  repeated AidingStateEnum valid_values = 2; // The valid values for the aiding state
}

// Position aiding configuration, defines which position input(s) to use for aiding
message AidingConfiguration{
  sonardyne.api.pub.common.Uid id         = 1; // Used to identify recipient instrument/module
  sonardyne.api.pub.common.Result result  = 2; // Populated by instrument when replying to a SetState indicating the success of the operation
  AidingState enable_gnss                 = 3; // Used to turn GNSS aiding on or off
  AidingState enable_xpos                 = 4; // Used to turn XPOS aiding on or off
  AidingState enable_usbl                 = 5; // Used to turn USBL aiding on or off
}


